packages/arb-validator-core/gobridge/executionChallenge.go

```diff
@@ -144,11 +146,13 @@ func (c *ExecutionChallenge) OneStepProof(
 	assertion *valprotocol.ExecutionAssertionStub,
 	proof []byte,
 ) error {
+	c.client.goEthMutex.Lock()
+	defer c.client.goEthMutex.Unlock()
 	valprotocol.ExecutionPreconditionHash(precondition.BeforeHash, precondition.TimeBounds, precondition.BeforeInbox.Hash())
 	precondition.Hash()
 
 	matchHash := valprotocol.ExecutionDataHash(1, precondition.Hash(), assertion.Hash())
-	if !c.client.challenges[c.contractAddress].challengerDataHash.Equals(matchHash) {
+	if !c.client.challenges[c.contractAddress].challengeData.challengerDataHash.Equals(matchHash) {
```

Comment:
Unnecessary indirection to get `challengeData`

Reply:
fixed